multiraft/multiraft.go

```diff
@@ -989,8 +982,21 @@ func (s *state) propose(p *proposal) {
 		}
 	}
 	if !found {
-		// If we are not a member of the group, don't allow any proposals.
-		s.removePending(nil, p, ErrGroupDeleted)
+		// If we are no longer a member of the group, don't allow any additional
+		// proposals.
+		//
+		// If this is an initial proposal, go ahead and remove the command. If
+		// this is a re-proposal, there is a good chance that the original
+		// proposal will still commit even though the node has been removed from
+		// the group.
+		//
+		// Therefore, we will not remove the pending command until the group
+		// itself is removed, the node is re-added to the group
+		// (which will trigger another re-proposal that will succeed), or the
+		// proposal is committed.
+		if initialProposal {
```

Comment:
isn't `initialProposal` equal to the `ok` in `_, ok := g.pending[p.commandID]` and thus not required?